pkg/crc/api/api_windows.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+	"golang.org/x/sys/windows/svc/eventlog"
+)
+
+var elog debug.Log
+
+type crcDaemonService struct {
+	socketPath string
+}
+
+func (m *crcDaemonService) Execute(args []string, r <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+	crcApiServer, err := CreateApiServer(m.socketPath)
+	if err != nil {
+		elog.Error(1, fmt.Sprintf("Failed to start CRC daemon service: %v", err)) // nolint
+		return
+	}
+	// Windows servivce manager calls OnStart when starting the service
+	// and it calls this method, which the service manager expects to handle
+	// commands like stop, shutdown. crcApiServer.Serve is being called as a go routine
+	// since we don't want it block
+	go crcApiServer.Serve()
+	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
+
+	MainLoop(r, changes)
+
+	// Above MainLoop will return when the daemon service is stopped
+	changes <- svc.Status{State: svc.Stopped}
+	return
+}
+
+func MainLoop(r <-chan svc.ChangeRequest, changes chan<- svc.Status) {
+	for { // nolint
```

Comment:
can this be `while(running)` and toggle the boolean instead of using a return?

Reply:
```diff
func MainLoop(r <-chan svc.ChangeRequest, changes chan<- svc.Status) {
+       var running = true
-	for { // nolint
+       while(running) {
		select {
		case c := <-r:
			switch c.Cmd {
			case svc.Interrogate:
				changes <- c.CurrentStatus
			case svc.Stop, svc.Shutdown:
				elog.Info(1, "Shutting down CRC daemon service") // nolint
-				return
+                               running = false
			default:
				elog.Error(1, fmt.Sprintf("unexpected control request #%d", c)) // nolint
			}
		}
	}
+       return
}
```

Reply:
can `elog` be named `eventlog` ?

Reply:
No, there's already a import with that name..

https://github.com/code-ready/crc/pull/1221/files#diff-8e01d564e3fa24ba40438055729bc063R9

Reply:
> can this be `while(running)` and toggle the boolean instead of using a return?

@gbraad Updated! Had to remove the return from `MainLoop` as linter was complaining, anyways once the loop break it has to return, there;s no code after it..

Reply:
even in that case we can use aliases for those